Task to move 3 of  hk2 modules from hk2 to Glassfish

sumauppa opened the pull request:
Moving hk2-config, config-generator plugin, config-types and ctm-exmple modules to Glassfish code. This has been placed under nucleus/hk2.

romain-grecourt committed:
revert poms to 2.5.0-b37-SNAPSHOT

glassfishrobot committed:
[maven-release-plugin] prepare release 2.5.0-b37

glassfishrobot committed:
[maven-release-plugin] prepare for next development iteration

jwells131313 committed:
This is a change but one test fails.  I need to see how it works on
a clean branch so checking in like this

jwells131313 committed:
Test case found a legitimate error

jwells131313 committed:
Change so that all validation exceptions still are wrapped in MultiException
since people do catch ME and may not catch the ConstratintValidationExcetpion

glassfishrobot committed:
[maven-release-plugin] prepare release 2.5.0-b38

glassfishrobot committed:
[maven-release-plugin] prepare for next development iteration

jwells131313 committed:
Fix copyright

jwells131313 committed:
Add a test that includes a validation during an addRoot and also
update guice to 4.1.0

jwells131313 committed:
Fix issue concerning validation information

mtau11 committed:
permit a range of javax.validation / hibernate.validator versions

mtau11 committed:
Merge pull request #361 from javaee/validation_osgi_version

permit a range of javax.validation / hibernate.validator versions

jwells131313 committed:
Fix copyrights

jwells131313 committed:
Merge branch 'master' of github.com:javaee/hk2

jwells131313 committed:
Update version of javassist

jwells131313 committed:
Update version of jboss-logging

jwells131313 committed:
Classmate explicit version and jboss logging explict version

jwells131313 committed:
Upgrade version of woodstox

jwells131313 committed:
Update to latest version of spring context

jwells131313 committed:
Standardize how guice version is specified

jwells131313 committed:
Remove any dependency from hk2-xml to hk2-config inclusing any and all unit
tests related to that dependency

jwells131313 committed:
Code coverage, new tests and some dead code removed

jwells131313 committed:
More code coverage tests, more dead code removed

jwells131313 committed:
More negative cases for code coverage and one positive case for code coverage

Commit:
[maven-release-plugin] prepare release 2.5.0-b39

Commit:
[maven-release-plugin] prepare for next development iteration

romain-grecourt committed:
upload site javadocs from maven

Commit:
[maven-release-plugin] prepare release 2.5.0-b40

Commit:
[maven-release-plugin] prepare for next development iteration

Commit:
[maven-release-plugin] prepare release 2.5.0-b41

Commit:
[maven-release-plugin] prepare for next development iteration

jwells131313 committed:
update version of woodstox

romain-grecourt committed:
use jvnet-parent as parent for the BOM POM

glassfishrobot committed:
[maven-release-plugin] prepare release 2.5.0-b42

glassfishrobot committed:
[maven-release-plugin] prepare for next development iteration

romain-grecourt committed:
remove site-deploy from the release goals

romain-grecourt committed:
revert version to 2.5.0-b42-SNAPSHOT

glassfishrobot committed:
[maven-release-plugin] prepare release 2.5.0-b42

romain-grecourt committed:
revert version to 2.5.0-b42-SNAPSHOT

glassfishrobot committed:
[maven-release-plugin] prepare release 2.5.0-b42

glassfishrobot committed:
[maven-release-plugin] prepare for next development iteration

jwells131313 committed:
Remove osgi-resource-locator from hk2, it lives in hk2-extras repo

romain-grecourt committed:
add copyright-plugin configuration in bom/pom.xml

romain-grecourt committed:
remove site from master branch

romain-grecourt committed:
use an older version for the in-house plugins used during the build

mtau11 committed:
Fix annotation processing to handle TYPE_USE annotations

mtau11 committed:
Merge pull request #367 from javaee/javax.validation.2.0.0.CR2

Fix annotation processing to handle TYPE_USE annotations

jwells131313 committed:
Failing test with checkpoint

jwells131313 committed:
Fix copyright

jwells131313 committed:
Merge branch 'master' into ContraintValidation

jwells131313 committed:
Passing test using a validator to cross-validate values

jwells131313 committed:
Fix copyright

romain-grecourt committed:
use hk2.mvn.plugins.version for all in-house plugins

romain-grecourt committed:
move maven-plugins out of the default reactor, remove old modules that got added back during java.net migration

glassfishrobot committed:
[maven-release-plugin] prepare release 2.5.0-b43

glassfishrobot committed:
[maven-release-plugin] prepare for next development iteration

jwells131313 committed:
Make sure all classloading is done under the permissions of hk2-config

jwells131313 committed:
Fix copyright

romain-grecourt committed:
- add org.glassfish.hk2.external:javax.inject back to bom
- Create an aggregator for the maven-plugins that were moved out of the default reactor.
- This pom will be used to release all maven plugins at once.
- Update release.sh to release maven-plugins.
- Add scm to maven-plugins/pom.xml

romain-grecourt committed:
fix copyright for maven-plugins directory

glassfishrobot committed:
[maven-release-plugin] prepare release maven-plugins-2.5.0-b42

romain-grecourt committed:
update maven-release-plugin configuration

romain-grecourt committed:
Update release.sh
  - add a loop to delete multiple tags
  - cleanup

Commit:
[maven-release-plugin] prepare release 2.5.0-b44

Commit:
[maven-release-plugin] prepare for next development iteration

Commit:
[maven-release-plugin] prepare release 2.5.0-b45

Commit:
[maven-release-plugin] prepare for next development iteration

romain-grecourt committed:
Make maven-plugins the parent pom of all maven plugin.
Use hk2.version to drive the hk2 version used by the maven plugins.

romain-grecourt committed:
force update pom version to 2.5.0-b44-SNAPSHOT

Commit:
[maven-release-plugin] prepare release 2.5.0-b44

Commit:
[maven-release-plugin] prepare for next development iteration

romain-grecourt committed:
update hk2.version property in maven-plugins/pom.xml

romain-grecourt committed:
force update pom version to 2.5.0-b44-SNAPSHOT

romain-grecourt committed:
bump hk2.version used in maven-plugins

Commit:
[maven-release-plugin] prepare release 2.5.0-b44

Commit:
[maven-release-plugin] prepare for next development iteration

romain-grecourt committed:
force a different tag for maven-plugins

romain-grecourt committed:
force update version to 2.5.0-b44-SNAPSHOT

Commit:
[maven-release-plugin] prepare release 2.5.0-b44

Commit:
[maven-release-plugin] prepare for next development iteration

romain-grecourt committed:
remove deploy skip and install skip

Commit:
[maven-release-plugin] prepare release 2.5.0-b45

Commit:
[maven-release-plugin] prepare for next development iteration

romain-grecourt committed:
bump release-plugin + scm-provider combo

romain-grecourt committed:
[maven-release-plugin] prepare release maven-plugins-2.5.0-b44

romain-grecourt committed:
[maven-release-plugin] prepare for next development iteration

romain-grecourt committed:
force update pom version to 2.5.0-b44-SNAPSHOT

romain-grecourt committed:
use maven-scm-provider-gitexe:1.9.4

Commit:
[maven-release-plugin] prepare release 2.5.0-b44

Commit:
[maven-release-plugin] prepare for next development iteration

Commit:
[maven-release-plugin] prepare release maven-plugins-2.5.0-b44

Commit:
[maven-release-plugin] prepare for next development iteration

jwells131313 committed:
Ensure the RunLevelContext has a specific name so it can be found easily
amongst other Context implementations (although there were already other ways
to do this, but those did involve ParameterizedTypes, so, this way
is probably best)

jwells131313 committed:
Fix copyright

jwells131313 committed:
Move hk2-json under the hk2-xml umbrella

jwells131313 committed:
Skeleton for bean to pbuf feature

jwells131313 committed:
Flesh out the first working test, for a very very simple bean to see
if the strategy can even work for pbuf

jwells131313 committed:
Checkpoint with protobuf

jwells131313 committed:
Very first pbuf written out!

jwells131313 committed:
Fix copyright

jwells131313 committed:
Failing test for structured marshal

jwells131313 committed:
Handle marshalling children

jwells131313 committed:
failing test for unmarshalling pbuf

jwells131313 committed:
Basic read/write of basic types (String, int, long) works.  No arrays
or any other type currently would work.  But that's all sort of icing on
the cake

romain-grecourt committed:
exclude .pbuf from copyright duty

jwells131313 committed:
For possibly generating schemas or the like include the requireness
in the not-yet-finalized SPI

jwells131313 committed:
Translate new required field into the proto file

jwells131313 committed:
Failing test for various types

jwells131313 committed:
All simple java types work.  Not sure about arrays or sets yet

jwells131313 committed:
Ensure the XmlType has the correct values for proper ordering of fields

jwells131313 committed:
Ordering again.  I need some error detection for XmlType

jwells131313 committed:
Ensure packages work properly along with test

jwells131313 committed:
Remove dead code

jwells131313 committed:
Fix javadoc

jwells131313 committed:
Initial protocol buf generator which does not handle any types or
ordering as of now.  Simple test to follow

jwells131313 committed:
Test that verifies at least that the proto files are generated.  Not
a gr8 test, but I'll need more infrastructure to get this to work
better

jwells131313 committed:
Most types are now converted properly for pbuf

jwells131313 committed:
Refactor to make it, like, better.  Only do the analysis once, let that
live past doImport and clean up the data structures

jwells131313 committed:
Ensure ordering via XmlType works

jwells131313 committed:
Better name for the annotation

jwells131313 committed:
Fix up a few things and ensure we can write out every type

jwells131313 committed:
Interop tests xml->json, xml->pbuf

jwells131313 committed:
More interop tests, they work gr8

jwells131313 committed:
Fix ordering, taking into account proper decapitalization of the fields

jwells131313 committed:
Required is not a thing in proto3, don't write it out

Commit:
[maven-release-plugin] prepare release 2.5.0-b45

Commit:
[maven-release-plugin] prepare for next development iteration

Commit:
[maven-release-plugin] prepare release maven-plugins-2.5.0-b45

Commit:
[maven-release-plugin] prepare for next development iteration

jwells131313 committed:
Tests for strings and integers and longs that are not set

jwells131313 committed:
Make the PBufUtilities method actually be idempotent

Commit:
[maven-release-plugin] prepare release 2.5.0-b46

Commit:
[maven-release-plugin] prepare for next development iteration

Commit:
[maven-release-plugin] prepare release maven-plugins-2.5.0-b46

Commit:
[maven-release-plugin] prepare for next development iteration

jwells131313 committed:
Fix locks in pbuf and make sure "is" works properly as well as the
declared scalar types like Boolean and Integer

jwells131313 committed:
@Comment feature for adding comments to generated proto files

Commit:
[maven-release-plugin] prepare release 2.5.0-b47

Commit:
[maven-release-plugin] prepare for next development iteration

Commit:
[maven-release-plugin] prepare release maven-plugins-2.5.0-b47

Commit:
[maven-release-plugin] prepare for next development iteration

jwells131313 committed:
Add in test for streaming pbufs and change marshal/demarshal to include
the size field in the pbuf output/input stream

Commit:
[maven-release-plugin] prepare release 2.5.0-b48

Commit:
[maven-release-plugin] prepare for next development iteration

Commit:
[maven-release-plugin] prepare release maven-plugins-2.5.0-b48

Commit:
[maven-release-plugin] prepare for next development iteration

jwells131313 committed:
Fix the parser to handle other beans using the same child

jwells131313 committed:
OneOf feature with failing test

jwells131313 committed:
Found a bug in the defaulting of values in addRoot.  Added in a test
at the source (hk2-xml main) and then also found a bug in the defaulting
test (hadn't been testing byte defaults by accident)

jwells131313 committed:
Checkpoint progress, the protobuf is being properly generated (I think)
at this point.  There is an issue with settedness in the unmarshaller

jwells131313 committed:
Test works!  Now for the second part, doing it in the generated .proto!

jwells131313 committed:
oneof works in the proto generator.  Nice!

jwells131313 committed:
A better test at least checks that the output is the same all the time.
This will obviously be a pain when new things are added, but is also worth
it.

jwells131313 committed:
Modify the generator to use protobuf style names rather than java names

jwells131313 committed:
Add in the generated java files from the protoc compiler

Commit:
[maven-release-plugin] prepare release 2.5.0-b49

Commit:
[maven-release-plugin] prepare for next development iteration

jwells131313 committed:
Test which verifies the generated protos match the dynamically generated
protos now works.  Had to do a few things (like change the names and
set of labels and stuff) but now it basically works fine

jwells131313 committed:
Merge branch 'master' of github.com:javaee/hk2

jwells131313 committed:
Wow! Minor bug in protocol buffers regarding the settedness of boolean
fields (when set to false).  This causes the hk2 byte stream to be
slightly different from the protocol buffer byte stream, but the data
values are still the same.  Only the settedness of false valued boolean
fields might be wrong

Commit:
[maven-release-plugin] prepare release 2.5.0-b50

Commit:
[maven-release-plugin] prepare for next development iteration

Commit:
[maven-release-plugin] prepare release maven-plugins-2.5.0-b49

Commit:
[maven-release-plugin] prepare for next development iteration

jwells131313 committed:
Make the stub processor use the proper package so the output file
goes in the correct place

jwells131313 committed:
Add some reflection-like capability to XmlHk2ConfigurationBean

jwells131313 committed:
Add in copyrights

jwells131313 committed:
Copyrights into the generated java files

Commit:
[maven-release-plugin] prepare release 2.5.0-b51

Commit:
[maven-release-plugin] prepare for next development iteration

Commit:
[maven-release-plugin] prepare release 2.5.0-b52

Commit:
[maven-release-plugin] prepare for next development iteration

Commit:
[maven-release-plugin] prepare release maven-plugins-2.5.0-b50

Commit:
[maven-release-plugin] prepare for next development iteration

jwells131313 committed:
No locks around marshal/unmarshal, it is innappropriate here

Commit:
[maven-release-plugin] prepare release 2.5.0-b53

Commit:
[maven-release-plugin] prepare for next development iteration

jwells131313 committed:
Amazing test bug not noticed for a long time.  Caused intermittent pbuf
failures

jwells131313 committed:
Validate the **** out of the XmlType propOrder field at runtime

Commit:
[maven-release-plugin] prepare release 2.5.0-b54

Commit:
[maven-release-plugin] prepare for next development iteration

Commit:
[maven-release-plugin] prepare release maven-plugins-2.5.0-b51

Commit:
[maven-release-plugin] prepare for next development iteration

jwells131313 committed:
Add Greedy JIT resolver and allow it to be enabled with SLU

jwells131313 committed:
XmlParser services should be LOCAL and we make the stream one the default
by giving it a Rank of 1 rather than making the others have the rank
of -1, makes more sense

Commit:
[maven-release-plugin] prepare release 2.5.0-b55

Commit:
[maven-release-plugin] prepare for next development iteration

Commit:
[maven-release-plugin] prepare release maven-plugins-2.5.0-b52

Commit:
[maven-release-plugin] prepare for next development iteration

jwells131313 committed:
Failing test for streaming marshal of array data

jwells131313 committed:
Handle array non-children properly in streaming marshal

Commit:
[maven-release-plugin] prepare release 2.5.0-b56

Commit:
[maven-release-plugin] prepare for next development iteration

Commit:
[maven-release-plugin] prepare release maven-plugins-2.5.0-b53

Commit:
[maven-release-plugin] prepare for next development iteration

sumauppa committed:
Keeping only hk2 modules to be moved from HK2

sumauppa committed:
Placed HK2 code in proper directory of Glassfish code

sumauppa committed:
mvn clean install is working except config-generator changes.

sumauppa committed:
Modified HK2 version.

sumauppa committed:
removed extra files created.

sumauppa committed:
restore files

sumauppa committed:
undo surefire-plugin version

sumauppa committed:
hibernate version to use as GF.

sumauppa committed:
Copyright changes

sumauppa committed:
pom merge

sumauppa committed:
commented appserverstartuptest

sumauppa committed:
copyright license file changes

sumauppa committed:
Added packaging to HK2 elements

sumauppa committed:
Removed config-generator packaging element

sumauppa committed:
Fixed config-generator plugin issue.

sumauppa committed:
Fixed hk2 xml parent pom

sumauppa committed:
Merge remote-tracking branch 'upstream/master' into hk2_test

sumauppa committed:
fixes config-generator

sumauppa committed:
Fixed syntax

sumauppa committed:
Adding maven-jar config to test

sumauppa committed:
excluded hk2 config and config types from console devtests pom to test

sumauppa committed:
Revert hk2-config and devtests pom changes that had done for testing.

sumauppa committed:
Merge remote-tracking branch 'upstream/master' into hk2_test

sumauppa committed:
adding configuration to maven-jar-plugin of hk2-config

sumauppa committed:
Removed test-jar from ctm example.

sumauppa committed:
Merge remote-tracking branch 'upstream/master' into hk2_test

sumauppa committed:
added test-jar back.

sumauppa committed:
Added findbugs profile to hk2-config.

sumauppa committed:
Merge remote-tracking branch 'upstream/master' into hk2_test

sumauppa committed:
Fixed findbugs warnings

sumauppa committed:
Merge remote-tracking branch 'upstream/master' into hk2_test

sumauppa commented:
@glassfishrobot Run CI tests please

yaminikb commented on nucleus/pom.xml:
```diff
@@ -138,8 +138,8 @@
         <saaj-api.version>1.4.0</saaj-api.version>
-        <hk2.version>2.5.0-b44</hk2.version>
-        <hk2.plugin.version>2.5.0-b44</hk2.plugin.version>
+        <hk2.version>2.5.0-b57</hk2.version>
```
The latest release of HK2 is 2.5.0-b60, you should be using that.

sumauppa replied:
I am using the version after removing HK2 modules from HK2 code. May be we can create an issue and update the hk2 versions. What do you think?

yaminikb commented on nucleus/pom.xml:
```diff
@@ -138,8 +138,8 @@
-        <hk2.version>2.5.0-b44</hk2.version>
-        <hk2.plugin.version>2.5.0-b44</hk2.plugin.version>
+        <hk2.version>2.5.0-b57</hk2.version>
+        <hk2.plugin.version>2.5.0-b54</hk2.plugin.version>
```
Please clarify why we are using 2 versions of HK2

sumauppa replied:
If there are any updates in HK2 code and not in plugins, they only release hk2 and plugins will retain the same old hk2 version. So, we have different versions for hk2 and hk2 plugins.

yaminikb commented on nucleus/core/kernel/src/test/java/com/sun/enterprise/v3/server/AppServerStartupTest.java:
```diff
@@ -254,6 +255,7 @@ public void afterTest() {
      * the run level services are constructed and destroyed at the proper
      * run levels.
      */
+    @Ignore
```
1. Add proper comments for why these tests are commented. 
2. Raise issue in GlassFish or HK2 for these failures.

sumauppa replied:
These tests are failing after moving HK2 modules from HK2 to Glassfish. I will raise an issue once the changes are committed.  Will add a comment.

sumauppa committed:
Added comment to AppServerStartupTest

sumauppa commented:
CI tests are success, except 11 findbugs warnings in hk2-config module.

sumauppa committed:
Updated hk2 versions

sumauppa commented:
@glassfishrobot Run CI tests please

sumauppa committed:
Updated a comment with Issue number

romain-grecourt commented:
1. Why do we need to migrate ctm-example ?
With this change, the old HK2 config framework will be buried inside the GlassFish workspace, I don't think it needs example. If it we were keeping, I would have it under hk2/config-examples/ctm/

2. Unless Maven supports it with recent versions, it is not possible to build plugin and use as part of the same reactor (i.e same Maven invocation) ; this is sometimes possible but these are tricky glitches... You should consider releasing this plugin separately and make it generic so that it can use the right hk2-config from the build.

3. Why are you duplicating plugin configuration in all these pom.xml files ?

sumauppa commented:
1)ctm-example is basically dependent on hk2-config and config-generator. So, we had to move this example to Glassfish. 
2)Initially had discussed with Yamini and then we thought to go ahead with this approach. Will Discuss again with her.
3)As we had to build the plugin in the same reactor, had to remove config-generator dependency from nucleus parent and placed it in required modules. Otherwise, we run into cycllic dependecies under nucleus/hk2.

sumauppa committed:
Moved nucleues/hk2/examples to nucleus/hk2/config-examples

sumauppa committed:
Merge remote-tracking branch 'upstream/master' into hk2_test

sumauppa commented:
@glassfishrobot Run CI tests please

sumauppa commented:
1) Changed the direcory name from examples to config-examples (nucleus/hk2/config-examples). Ran CI tests after that.
2) As this is only used in Glassfish, we decided to keep in the same reactor. We made sure that this plugin is built first and ran all CTS and CI tests to make sure nothing is broken.
3)As plugin is in the same reactor, there would be cyclic dependency. Hence, removed that plugin from nucleus pom and placed it under the modues, wherever it is required.

jwells131313 reviewed (dismissed):
I'm really happy this is getting done!  Thanks very much for doing it

sumauppa committed:
Added hk2 under nucleus/parent/hk2 and made the dependency in a way that nucleus depends on nucleus-parent.

sumauppa committed:
nucleus pom changes

sumauppa committed:
added extra space

sumauppa committed:
Removed config-generator from appserver pom.

sumauppa committed:
extra space issue.

sumauppa committed:
extra space issue.

sumauppa committed:
extra space issue.

sumauppa commented:
@glassfishrobot Run CI tests please

sumauppa committed:
added relative path

sumauppa committed:
nucleus parent change

sumauppa committed:
Kept config-generator in nucleus pom.

sumauppa committed:
moving back hk2 to nucleus/hk2

sumauppa commented:
@glassfishrobot Run CI tests please

sumauppa commented:
@glassfishrobot Run CI tests please

sumauppa committed:
Added parent also in nucleus pom.

sumauppa commented:
@glassfishrobot Run CI tests please

sumauppa commented:
@glassfishrobot Run CI tests please

sumauppa committed:
Merge remote-tracking branch 'upstream/master' into hk2_test

sumauppa committed:
removed ctm-example

sumauppa committed:
copyright update

sumauppa commented:
@glassfishrobot Run CI tests please

romain-grecourt reviewed (dismissed):
Approved.
Is there any workarounds leftover ?

sumauppa commented:
Yes. We use in nucleus/hk2/pom.xml, overriding config-generator.

sumauppa committed:
resolved merge conflict.

sumauppa committed:
space issue

sumauppa commented:
@glassfishrobot Run CI tests please

sumauppa committed:
copied all properties to nucleus

sumauppa commented:
Moved all properties to nucleus parent.

sumauppa commented:
@glassfishrobot Run CI tests please

yaminikb reviewed (approved):
Looks good.
